### PR TITLE
Create Socket Based Control Example

### PR DESCRIPTION
**Pull request by AkhilDeo**
Create an example for controlling surgical_robotics_challenge PSMs wirelessly via socket. Includes detailed README for usage.

### CODE REVIEW

File: scripts/surgical_robotics_challenge/examples/socket_based_control/socket_based_control.py [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+import sys
+import signal
+
+
+UDP_IP = socket.gethostbyname(socket.gethostname())
+UDP_PORT = 8080
+sock = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
+sock.bind(("", UDP_PORT))
+
+_client = Client()
```
Comment (adnanmunawar) [inline scripts/surgical_robotics_challenge/examples/socket_based_control/socket_based_control.py:64]: I think we should replace the references of Client with SimulationManager class. Please see an example here (https://github.com/surgical-robotics-ai/surgical_robotics_challenge/blob/master/scripts/surgical_robotics_challenge/examples/interface_via_method_api.py#L33)

File: scripts/surgical_robotics_challenge/examples/socket_based_control/socket_based_control.py [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+_client.connect()
+print(_client.get_obj_names())
+w = _client.get_world_handle()
+w.reset_bodies()
+psm1 = PSM(_client, 'psm1')
```
Comment (adnanmunawar) [inline scripts/surgical_robotics_challenge/examples/socket_based_control/socket_based_control.py:69]: The PSM and ECM class can take in the SimulationManager object that I mentioned above

**Review by adnanmunawar — COMMENTED** [review]
Please see the minor comments that I left in there

File: scripts/surgical_robotics_challenge/examples/socket_based_control/socket_based_control.py (hunk 1 above)
Comment (AkhilDeo) [reply scripts/surgical_robotics_challenge/examples/socket_based_control/socket_based_control.py]: Makes sense to me, done (check new commits)

**Comment by AkhilDeo** [comment]
> Please see the minor comments that I left in there

Addressed

**Review by adnanmunawar — APPROVED** [review]

**Pull request merged**